Wordle/GameManager.py

```diff
@@ -1,25 +1,42 @@
-from typing import Optional
+from contextlib import asynccontextmanager
 
+from discord import TextChannel
+
+from Wordle.Canvas import Canvas
 from Wordle.Game import Game
+from Wordle.Lock import Lock
+from Wordle.Store import Store
 
 
 class GameManager:
-    def __init__(self):
-        self.games: dict = {}
+    def __init__(self, backend: Store):
+        self.store: Store = backend
+        self.canvas: Canvas = Canvas()
+
+    def create_game(self, word_length: int, mode: str) -> Game:
+        return Game(word_length=word_length, mode=mode, canvas=self.canvas)
+
+    @asynccontextmanager
+    async def lock(self, channel: TextChannel) -> Lock:
+        """ raises LockNotFoundError """
+        async with self.store.lock(channel.guild.id, channel.id):
```

Comment:
Referencing the `guild` throughout this file has broken the bot in DMs, as DMs are guildless channels. The bot working in DMs is a feature I'd like to keep if possible. Playing around with potential fixes, it looks like replacing each instance of `channel.guild.id` in the file with `channel.guild.id if not isinstance(channel, DMChannel) else 0` appears to work. That solution feels a wee bit hacky though, so gonna keep poking around for something a little cleaner.

Reply:
Good catch.